src/components/RoomHeaderAvatars.js

```diff
@@ -25,14 +28,31 @@ function RoomHeaderAvatars(props) {
 
     if (props.icons.length === 1) {
         return (
-            <Avatar
-                source={props.icons[0].source}
-                imageStyles={[styles.avatarLarge]}
-                fill={themeColors.iconSuccessFill}
-                size={CONST.AVATAR_SIZE.LARGE}
-                name={props.icons[0].name}
-                type={props.icons[0].type}
-            />
+            <AttachmentModal
+                headerTitle={props.icons[0].name}
+                source={UserUtils.getFullSizeAvatar(props.icons[0].source, props.icons[0].id)}
+                isAuthTokenRequired
```

Comment:
@spcheema @aimane-chnaif

I have a query regarding the `isAuthTokenRequired` flag set to true for the `AttachmentModal` component. This flag's setting becomes particularly interesting when compared with the `Avatar` component's behavior. Both `AttachmentModal` and `Avatar` utilize the same image source (`props.icons[0].source`), yet `Avatar` manages to load its image without requiring authentication. 

This leads me to wonder if there are specific scenarios where authentication is indeed necessary for loading an avatar image. Understanding this is crucial because if authentication is not required in these cases, we might be exposing our authentication tokens to external addresses unnecessarily. 

Furthermore, it’s important to consider future updates, particularly regarding the Image component's adaptation to use header-based authentication. Such changes can introduce complexities due to browser security features like CORS (Cross-Origin Resource Sharing). Since avatar sources are not hosted on the same origin, this could potentially lead to CORS errors, disrupting the user experience.

Could you provide insights on the necessity of the `isAuthTokenRequired` flag in this context? Understanding the rationale behind this decision will help us ensure both secure and efficient handling of image sources across different components.

Reply:
I'm also interested - AFAIK the auth tokens were only needed for chat attachments, since these obviously need to be private (only accessible by the participants of a report where the chat attachment was sent). I don't know of / remember a reason we would need any avatars to be private & require auth tokens to fetch